Fix norm decomp when dtype is passed in

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #89508

Fix for https://github.com/pytorch/torchdynamo/issues/1889. The wrapper was doing a downcast even when the dtype was explicitly passed in.


cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

eellison committed:
Fix norm decomp when dtype is passed in

[ghstack-poisoned]

eellison committed:
Update on "Fix norm decomp when dtype is passed in"


Fix for https://github.com/pytorch/torchdynamo/issues/1889. The wrapper was doing a downcast even when the dtype was explicitly passed in.


cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

[ghstack-poisoned]

eellison committed:
Update on "Fix norm decomp when dtype is passed in"


Fix for https://github.com/pytorch/torchdynamo/issues/1889. The wrapper was doing a downcast even when the dtype was explicitly passed in.


cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

[ghstack-poisoned]

eellison committed:
Update on "Fix norm decomp when dtype is passed in"


Fix for https://github.com/pytorch/torchdynamo/issues/1889. The wrapper was doing a downcast even when the dtype was explicitly passed in.


cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

[ghstack-poisoned]

eellison committed:
Update on "Fix norm decomp when dtype is passed in"


Fix for https://github.com/pytorch/torchdynamo/issues/1889. The wrapper was doing a downcast even when the dtype was explicitly passed in.


cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

[ghstack-poisoned]

eellison committed:
Update on "Fix norm decomp when dtype is passed in"


Fix for https://github.com/pytorch/torchdynamo/issues/1889. The wrapper was doing a downcast even when the dtype was explicitly passed in.


cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

[ghstack-poisoned]

eellison commented:
@pytorchbot merge

ngimel commented on torch/_decomp/decompositions.py:
```diff
@@ -1775,17 +1775,23 @@ def log_sigmoid_forward(self: Tensor) -> Tuple[Tensor, Tensor]:
+    if dtype:
+        return torch.linalg.vector_norm(self, p, dim, keepdim, dtype=dtype)
+
+    computation_dtype, result_dtype = utils.elementwise_dtypes(
```
do you need this? Just calling `vector_norm` should be enough. Alternatively, we can switch to `norm` decomposition from `_refs` (and it's just calling `torch.linalg.vector_norm`, passing through dtype arg).

eellison replied:
i think it's still good to model the dtype conversion so inductor can fuse it

ngimel replied:
Inductor gets fusable dtype conversions from `linalg_vector_norm` decomposition that is uses, so I really think there's no point in this decomp, we should just add registration to the one in `_refs` that does the same thing.

eellison committed:
Update on "Fix norm decomp when dtype is passed in"


Fix for https://github.com/pytorch/torchdynamo/issues/1889. The wrapper was doing a downcast even when the dtype was explicitly passed in.


cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

[ghstack-poisoned]

eellison commented:
@pytorchbot merge -f "flakey failure"
